### PR TITLE
[Refactoring] Mark fields as final if they are not reassigned

### PR DESCRIPTION
**Pull request by Shadow-Devil**
## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/options/IndentFormattingOptions.java [hunk 1]
```diff
@@ -63,7 +63,7 @@ public static class IndentFormattingOptionsBuilder {
         private static final String CONTINUATION_INDENT_SIZE = "continuationIndentSize";
         private int indentSize = getDefaultInt(FormatSection.INDENT, INDENT_SIZE);
         private int continuationIndentSize = getDefaultInt(FormatSection.INDENT, CONTINUATION_INDENT_SIZE);
-        private String wsCharacter = " ";
+        private static final String wsCharacter = " ";
```
Comment (SasinduDilshara) [inline misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/options/IndentFormattingOptions.java:66]: if we define a variable `static final` we need to make variable name as capital.
`ex:- WS_CHARACTER`

File: tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/utils/RuntimeApi.java [hunk 2]
```diff
@@ -42,7 +42,7 @@ public class RuntimeApi {
 
     private static final Module objectModule = new Module("testorg", "runtime_api", "1");
     private static final Module recordModule = new Module("testorg", "runtime_api", "1");
-    private static Module errorModule = new Module("testorg", "runtime_api", "1");
+    private static final Module errorModule = new Module("testorg", "runtime_api", "1");
```
Comment (SasinduDilshara) [inline tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/utils/RuntimeApi.java:45]:
```suggestion
    private static final Module ERROR_MODULE = new Module("testorg", "runtime_api", "1");
```

**Comment by SasinduDilshara** [comment]
Other than https://github.com/ballerina-platform/ballerina-lang/pull/43028#discussion_r1665101857 LGTM

**Review by chiranSachintha — DISMISSED** [review]

**Review by SasinduDilshara — APPROVED** [review]

**Comment by NipunaRanasinghe** [comment]
@MaryamZi it seems this PR got closed due to inactivity. So we might need to re-open this and continue reviewing?

**Comment by Shadow-Devil** [comment]
Failing testcase in Ballerina Distribution:
```
Output Verification 'time-formatting-and-parsing'
Failed due to output did not match. Line 3: Civil string representation: 2021-04-12T23:20:50.520+05:30
WARNING: This command is deprecated and will be removed in a future release. Use `bal new .` instead
Building example 'time-formatting-and-parsing'
Created new package 'test'.
Compiling source
	runner/test:0.1.0

Running Tests

	test

	testFunc: has failed.

error: there are test failures

		[fail] testFunc:

		    error {ballerina/test:0}TestError ("Assertion Failed!
			 
			expected: 'Civil string representation: 2021-04-12T17:50:50.520Z'
			actual	: 'Civil string representation: 2021-04-12T23:20:50.520+05:30'
			 
			Diff	:
			
			--- actual
			+++ expected 
			 
			 @@ -1,1 +1,1 @@ 
			 
			 -Civil string representation: 2021-04-12T23:20:50.520+05:30
			+Civil string representation: 2021-04-12T17:50:50.520Z
			")
				callableName: createBallerinaError moduleName: ballerina.test.0 fileName: assert.bal lineNumber: 41
				callableName: assertEquals moduleName: ballerina.test.0 fileName: assert.bal lineNumber: 109
				callableName: testFunc moduleName: runner.test$test.0.tests.time_formatting_and_parsing_test fileName: tests/time_formatting_and_parsing_test.bal lineNumber: 79
				callableName: testFunc$lambda0$ moduleName: runner.test$test.0.tests.test_execute-generated_1 fileName: tests/test_execute-generated_1.bal lineNumber: 4
			


		0 passing
		1 failing
		0 skipped

		Test execution time : 0.145s
```
This is not related to this PR since it also happened in #42955

**Review by NipunaRanasinghe — DISMISSED** [review]
Bindgen tool related changes LGTM

**Review by gimantha — APPROVED** [review]

**Pull request merged**